custom_components/tahoma/state_sensor.py

```diff
@@ -0,0 +1,167 @@
+"""Support for TaHoma sensors."""
+
+from dataclasses import dataclass
+from typing import Any, Callable, Optional, Union
+
+from homeassistant.components import sensor
+from homeassistant.const import (
+    DEVICE_CLASS_ILLUMINANCE,
+    LIGHT_LUX,
+    PERCENTAGE,
+    SIGNAL_STRENGTH_DECIBELS,
+    VOLUME_FLOW_RATE_CUBIC_METERS_PER_HOUR,
+    VOLUME_LITERS,
+)
+from homeassistant.helpers.entity import Entity
+
+from .coordinator import TahomaDataUpdateCoordinator
+from .tahoma_entity import TahomaEntity
+
+
+@dataclass
+class StateDescription:
+    """Class to describe a sensor."""
+
+    key: str
+    name: str
+    icon: Optional[str] = None
+    unit: Union[None, str, Callable[[dict], str]] = None
+    value: Callable[[Any], Any] = lambda val: val
+    device_class: Optional[str] = None
+    default_enabled: bool = True
+
+
+SUPPORTED_STATES = [
+    StateDescription(
+        key="core:BatteryState",
+        name="Battery",
+        unit=PERCENTAGE,
+        device_class=sensor.DEVICE_CLASS_BATTERY,
+    ),
+    StateDescription(
```

Comment:
I would love to use `SensorEntityDescription`, see https://github.com/home-assistant/core/pull/53890/files for an example.

We would still need to create our own `OverkizSensorDescription` since we need some additional things like the `value` transformer, but that is fine.

Reply:
Oh clearly that’s something we should do with this PR. I totally missed this new class.

Reply:
Done.